pkg/client/query/interface.go

```diff
@@ -0,0 +1,16 @@
+package query
+
+// ParamsCache is an interface for a simple in-memory cache implementation for query parameters.
+type ParamsCache[T any] interface {
+	Get() (T, bool)
+	Set(T)
+	Clear()
+}
+
+// KeyValueCache is an interface for a simple in-memory key-value cache implementation.
+type KeyValueCache[V any] interface {
```

Comment:
@bryanchriswhite and I spent a lot of time designing a more general purpose interface in.

I think we should merge that one in and leverage it here.

[1] https://github.com/pokt-network/poktroll/pull/994/files

Reply:
This interface is consistent with what I was calling `QueryCache[T any]` in #994. It should be quite straightforward to refactor #994 to use this instead. I see `KeyValueCache[V any]` as a subsequent iteration of `QueryCache[T any]` which includes generalizing the name.

Reply:
This PR and PR #994 have very similar interfaces and will ultimately be merged.

I considered building on top of #994 but decided not to, as this PR does not aim to provide:

* Granular (per-key) cache invalidation
* Time-based cache invalidation (it only clears the whole cache when a new block is observed)
* Access to historical values, which is out of scope and would imply adding new functionality (or not using the implementation)

I can put in the effort to rebase it on top of #994 if you and @bryanchriswhite think it would be worthwhile.

Reply:
I left a comment on discord, but I feel strongly that we should finish #994 and rebase on top of it.

Now is the time to do this right.

<img width="1111" alt="Screenshot 2025-02-11 at 4 31 05 PM" src="https://github.com/user-attachments/assets/8de639a5-abd1-4e06-8513-f8b8b2ca49bb" />